src/main/java/com/howtank/streams/client/StreamClientHttp.java

```diff
@@ -54,54 +56,154 @@ public class StreamClientHttp implements StreamClient {
+        Objects.requireNonNull(streamId);
+        Objects.requireNonNull(referenceDate);
+        validateCountGreaterThanZero(count);
+        return getStreamMessages(streamId, referenceDate.getTime(), count, "both");
     }
 
     @Override
-    public List<StreamMessage> getStreamMessagesAroundDate(String streamId, Date referenceDate, int count) {
-        return null;
+    public void publishStreamMessage(String streamId, String content) throws HowtankApiException {
+        Objects.requireNonNull(streamId);
+        Objects.requireNonNull(content);
+        List<NameValuePair> params = RequestParams.builder()
+                .addIfPresent(RequestParameters.STREAM_ID, streamId)
+                .addIfPresent(RequestParameters.CONTENT, content)
+                .addIfPresent(RequestParameters.LOCAL_ID, (new Date()).getTime())
+                .addIfPresent(RequestParameters.TYPE, "group_chat")
+                .build();
+
+        AddStreamMessageApiResponse response = performGet(ADD_STREAM_MESSAGE, AddStreamMessageApiResponse.class, params);
+        validateAcceptedResponse(response);
     }
 
     @Override
-    public void publishStreamMessage(String streamId, String content) {
+    public void pushStreamMessageNotification(String streamId, String content) throws HowtankApiException {
+        Objects.requireNonNull(streamId);
+        Objects.requireNonNull(content);
+        List<NameValuePair> params = RequestParams.builder()
+                .addIfPresent(RequestParameters.STREAM_ID, streamId)
+                .addIfPresent(RequestParameters.CONTENT, content)
+                .addIfPresent(RequestParameters.LOCAL_ID, (new Date()).getTime())
+                .addIfPresent(RequestParameters.TYPE, "notification")
+                .build();
+
+        AddStreamMessageApiResponse response = performGet(PUSH_STREAM_MESSAGE, AddStreamMessageApiResponse.class, params);
+        validateAcceptedResponse(response);
+    }
 
+    private void validateCountGreaterThanZero(int count) {
+        if (count <= 0) {
+            throw new IllegalArgumentException("count should be greater than 0");
+        }
     }
 
-    @Override
-    public void pushStreamMessageNotification(String streamId, String content) {
+    private void validateAcceptedResponse(ApiResponse response) throws HowtankApiException {
+        if (!response.isAccepted()) {
+            throw new HowtankApiException(response.getError(), response.getReason());
+        }
+    }
+
+    private List<StreamMessage> getStreamMessages(String streamId, Long timestamp, int count, String direction) throws HowtankApiException {
+        List<NameValuePair> params = RequestParams.builder()
+                .addIfPresent(RequestParameters.STREAM_ID, streamId)
+                .addIfPresent(RequestParameters.DIRECTION, direction)
+                .addIfPresent(RequestParameters.TIMESTAMP, timestamp)
+                .addIfPresent(RequestParameters.COUNT, count)
```

Comment:
Note:
Currently, the howtank-web is not supporting get stream messages with a specific count. It always use STREAM_MESSAGES_PAGE_SIZE = 50 instead. 

We need to work on the babhelp-web component to support that.